### PR TITLE
HV-980 Exposing value on property nodes of constraint violation paths

### PR DESCRIPTION
**Pull request by gunnarmorling**
@hferentschik If you have a minute, can you take a look at that one? This provides a new HV-specific node interface `PropertyNode` with a method `getValue()` returning the value of the represented property. I.e. it's not only for sets but for all property nodes. If you agree with the general solution, I'll push another commit for the doc update. Thx!

### CODE REVIEW

**Comment by gunnarmorling** [comment]
Pushed one more commit for the doc update.

File: engine/src/main/java/org/hibernate/validator/path/PropertyNode.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.path;
+
+/**
+ * Node representing a property, providing Hibernate Validator specific functionality.
+ *
+ * @author Gunnar Morling
+ */
+public interface PropertyNode extends javax.validation.Path.PropertyNode {
```
Comment (hferentschik) [inline engine/src/main/java/org/hibernate/validator/path/PropertyNode.java]: nice to create the public class for this (compared to just exposing something internal). I think we get really good at this.

File: engine/src/test/java/org/hibernate/validator/test/internal/engine/path/NodeImplTest.java [hunk 2]
```diff
@@ -181,6 +189,95 @@ public void testToMapCascadeValidate() {
...
+		assertNumberOfViolations( constraintViolations, 1 );
+		assertCorrectPropertyPaths( constraintViolations, "apartments[].resident.name" );
+
+		Path path = constraintViolations.iterator().next().getPropertyPath();
+		Iterator<Path.Node> nodeIterator = path.iterator();
+
+		Path.Node node = nodeIterator.next();
+		assertEquals( node.getKind(), ElementKind.PROPERTY, "unexpected node kind" );
+		assertEquals( node.getName(), "apartments" );
+		assertEquals( node.as( PropertyNode.class ).getValue(), new Apartment( new Person( "Bob" ) ) );
```
Comment (hferentschik) [inline engine/src/test/java/org/hibernate/validator/test/internal/engine/path/NodeImplTest.java]: I think I would have created the apartment up at the top and added it to the building and used it for the assertion here. But that's a matter of taste manybe.

File: engine/src/main/java/org/hibernate/validator/internal/engine/path/NodeImpl.java [hunk 3]
```diff
@@ -78,6 +80,7 @@ public static NodeImpl createPropertyNode(String name, NodeImpl parent) {
 				null,
 				ElementKind.PROPERTY,
 				EMPTY_CLASS_ARRAY,
+				null,
```
Comment (hferentschik) [inline engine/src/main/java/org/hibernate/validator/internal/engine/path/NodeImpl.java]: gets a bit nasty with all this parameters which gets set to 'null' depending on the use-case. Not sure how one could improve that!?

**Comment by hferentschik** [comment]
Nice. This really grows on me. Might be even something for BV 1.2.

**Comment by hferentschik** [comment]
Merged.

**Pull request closed without merging**